src/core/control/RecentManager.cpp

```diff
@@ -230,6 +244,17 @@ void RecentManager::updateMenu() {
             xoj_files.emplace_back(recent);
         }
     }
+
+    if (pdf_files.empty() && xoj_files.empty()) {
+        GtkWidget* noFilesItem = gtk_menu_item_new_with_mnemonic(
+            _("No recent files"));
+        gtk_menu_shell_append(GTK_MENU_SHELL(menu), noFilesItem);
+        gtk_widget_set_visible(GTK_WIDGET(noFilesItem), true);
+        gtk_widget_set_sensitive(GTK_WIDGET(noFilesItem), false);
+        this->menuItemList.push_back(noFilesItem);
+        return;
```

Comment:
Is there a reason you removed the list clearing? Even if there are no recent files matching the types xopp or pdf, this GList might contain other types of files, I guess. Or did you get some warnings telling you that `items` was a nullptr?
```suggestion
        g_list_free_full(items, GDestroyNotify(gtk_recent_info_unref));
        return;
```

Reply:
> Is there a reason you removed the list clearing? Even if there are no recent files matching the types xopp or pdf, this GList might contain other types of files, I guess. Or did you get some warnings telling you that `items` was a nullptr?

I made some tests and it seemed to have no effect. Still, if you say that it could have, I may very well put it back in (and I just did, as a matter of fact!).

Reply:
The goal is to avoid a memory leak. The pointer `items` points to memory allocated via a gtk function. Said memory needs to be cleared before the destruction of the pointer itself.

Reply:
Where is the details of recent documents actually written? I am trying to clear it manually

Reply:
> Where is the details of recent documents actually written? I am trying to clear it manually

`~/.local/share/recently-used.xbel`